src/BaGet/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -22,6 +22,56 @@
 
 namespace BaGet.Extensions
 {
+
+    public static class ServiceProviderExtensions
+    {
+
+        public static DatabaseOptions GetDatabaseOptions(this IServiceProvider serviceProvider)
+        {
+            if (serviceProvider == null) throw new ArgumentNullException(nameof(serviceProvider));
+
+            var databaseOptions = serviceProvider.GetRequiredService<IOptions<BaGetOptions>>()
+                                    .Value
+                                    .Database ?? throw new ArgumentNullException(nameof(serviceProvider));

```

Comment:
` .Database ?? throw new ArgumentNullException(nameof(serviceProvider));` should be ` .Database;` as you check that `databaseOptions` isn't null on line 37.